src/main/java/com/expenses/ExpenseCliApp.java

```diff
@@ -0,0 +1,239 @@
+                            break;
+                        }
+                        for (Builder record : nLargestExpenses) {
+                            System.out.println(" Amount: " + record.getAmount() +
+                                    " |Data: " + record.getDate() +
+                                    " |Place: " + record.getPlace() +
+                                    " |Category: " + record.getCategory());
+                        }
+                        break;
+                    } catch (NullPointerException e) {
+                        e.printStackTrace();
+                    }
+                }
+                case "averageFromTo": {
+                    System.out.println("Enter 'from' date date dd-MM-yyyy");
+                    String entryDataFrom = scanner.nextLine();
+                    LocalDate dataFrom;
+                    try {
+                        dataFrom = LocalDate.parse(entryDataFrom, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+                    } catch (DateTimeException exception) {
+                        System.out.println("Entered data " + entryDataFrom + " id invalid.");
+                        break;
+                    }
+                    System.out.println("Enter 'to' date dd-MM-yyyy");
+                    String entryDataTo = scanner.nextLine();
+                    LocalDate dataTo = null;
+                    try {
+                        dataTo = LocalDate.parse(entryDataTo, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+                    } catch (DateTimeException exception) {
+                        System.out.println("Entered data " + entryDataTo + " id invalid.");
+                        break;
+                    }
+                    BigDecimal bigDecimal = service.averageOfExpensesInRangeOfTime(dataFrom, dataTo);
+                    System.out.println("Average of yours expenses in given time: " + bigDecimal);
+                    break;
+                }
+                case "averageInCategory": {
+                    System.out.println("Enter category");
+                    String categoryInput = br.readLine();
+                    BigDecimal bigDecimal = service.averageOfExpensesInCategory(categoryInput);
+
+                    System.out.println(bigDecimal);
+                    break;
+                }
+                case "mapAverage": {
+                    Map<String, BigDecimal> mapCategoryAndAverage = service.mapOfCategoryAndAverageOfExpenses();
+
+                    System.out.println(mapCategoryAndAverage.entrySet());
+                    break;
+                }
+                case "mapLargest": {
+                    Map<String, BigDecimal> mapLargest = service.mapOfCategoryAndLargestExpense();
+
+                    System.out.println(mapLargest.entrySet());
+
+                    break;
+                }
+
+                case "x": {
+                    System.out.println("Application closed");
+                    System.exit(1);
```

Comment:
Przekazywanie do `System.exit()` wartości innej niż 0, oznacza że program zakończył działanie błędem. Jest to mechanizm uwarunkowany dość starą "umową" w której systemy operacyjne sprawdzają wartość zwróconą przez zamykający się program.

Tutaj mam taką myśl - albo `System.exit()` który od razu zamknie aplikację, albo `break` - oba raczej nie są potrzebne. Ja bym chyba zostawił `break` bo sam program może chcieć jeszcze coś zrobić, np zapisać dane do pliku zanim się zamknie. `System.exit()` zamknie go natychmiastowo, bez wykonania dalszej części metody.